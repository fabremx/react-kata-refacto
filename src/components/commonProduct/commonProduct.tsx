import { ReactElement } from "react";
import baseProductStyles from "../baseProduct/baseProduct.module.scss";

interface Props {
  image: string;
  title: string;
}

export function CommonProduct({ image, title }: Props): ReactElement {
  return (
    <>
      <td className={baseProductStyles.photo} data-testid="photoCommonProduct">
        <img src={image} alt="product" />
      </td>

      <td className={baseProductStyles.name} data-testid="titleCommonProduct">
        {title}
      </td>
    </>
  );
}
