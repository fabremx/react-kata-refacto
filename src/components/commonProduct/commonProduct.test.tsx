import { render } from "@testing-library/react";
import { CommonProduct } from "./commonProduct";

describe("CommonProduct", () => {
  it("should render correctly the component", () => {
    // Given
    const tableRow = document.createElement("tr");
    // When
    const { asFragment } = render(
      <CommonProduct image={"Image"} title={"Title"} />,
      {
        container: document.body.appendChild(tableRow),
      }
    );
    // Then
    expect(asFragment()).toMatchSnapshot();
  });
});
