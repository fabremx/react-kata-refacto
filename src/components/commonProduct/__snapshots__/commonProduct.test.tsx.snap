// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonProduct should render correctly the component 1`] = `
<DocumentFragment>
  <img
    alt="product"
    src="Image"
  />
  Title
</DocumentFragment>
`;
